loki/transformations/tests/test_loop_blocking.py

```diff
@@ -0,0 +1,538 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+
+from math import floor
+
+import pytest
+import numpy as np
+
+from loki import available_frontends, Subroutine, pragmas_attached, find_driver_loops, Loop, fgen, \
+   ir, FindNodes, LoopRange, IntLiteral, jit_compile, clean_test, FindVariables, Array
+from loki.expression.parser import LokiEvaluationMapper
+from loki.transformations.loop_blocking import split_loop, block_loop_arrays, \
+    normalized_loop_range, iteration_number, iteration_index
+
+
+def get_pyrange(loop_range: LoopRange):
+    """
+    Converts a LoopRange of IntLiterals to a python range.
+    """
+    LEM = LokiEvaluationMapper()
+    if loop_range.step is None:
+        return range(LEM(loop_range.start), floor(LEM(loop_range.stop))+1)
+    else:
+        return range(LEM(loop_range.start), floor(LEM(loop_range.stop))+1, LEM(loop_range.step))
+
+
+@pytest.mark.parametrize('frontend', available_frontends())
+def test_normalized_loop_range(tmp_path, frontend):
+    """
+    Tests the num_iterations and normalized_loop_range functions.
+    """
+    for start in range(-10, 11):
+        for stop in range(start + 1, 50 + 1, 4):
+            for step in range(1, stop - start):
+                loop_range = LoopRange((start, stop, step))
+                pyrange = range(start, stop + 1, step)
+
+                normalized_range = normalized_loop_range(loop_range)
+                assert normalized_range.step is None, "LoopRange.step should be None in a normalized range"
+
+                normalized_start = LokiEvaluationMapper()(normalized_range.start)
+                assert normalized_start == 1, "LoopRange.start should be equal to 1 in a normalized range"
+
+                normalized_stop = floor(LokiEvaluationMapper()(normalized_range.stop))
+                assert normalized_stop == len(
+                    pyrange), "LoopRange.stop should be equal to the total number of iterations of the original LoopRange"
+
+
+@pytest.mark.parametrize('frontend', available_frontends())
+def test_iteration_number(tmp_path, frontend):
+    for start in range(-10, 11):
+        for stop in range(start + 1, 50, 4):
+            for step in range(1, stop - start):
```

Comment:
This lacks the `step=None` case, could easily be added like this:
```suggestion
            for step in itertools.chain([None], range(1, stop - start)):
```

Reply:
Yeah, that's a nice utility I wasn't aware of.